### PR TITLE
adding rocket chat notification to gha

### PR DESCRIPTION
**Pull request by rajpalc7**

### CODE REVIEW

File: .github/workflows/uninstall_install_sandbox.yaml [hunk 1]
```diff
@@ -28,15 +28,28 @@ jobs:
...
+        if: always()
+        with:
+          type:  ${{ job.status }}
+          job_name: '*Uninstall Traction Sandbox*'
+          mention: 'here'
+          mention_if: 'failure'
+          channel: '#ditp-gha-notifications'
+          url: ${{ secrets.ROCKETCHAT_WEBHOOK }}
+          commit: true
+          token: ${{ secrets.ROCKETCHAT_TOKEN }}
```
Comment (WadeBarnes) [inline .github/workflows/uninstall_install_sandbox.yaml:51]: `token: ${{ secrets.ROCKETCHAT_TOKEN }}` is not required if the `secrets.ROCKETCHAT_WEBHOOK` contains the token, which in our case it does.  Refer to your previous work with this action here; https://github.com/bcgov/von-bc-registries-audit/blob/main/.github/workflows/actions/deploy/action.yml#L83-L93

File: .github/workflows/uninstall_install_sandbox.yaml [hunk 2]
```diff
@@ -58,9 +71,22 @@ jobs:
...
+        if: always()
+        with:
+          type:  ${{ job.status }}
+          job_name: '*Install Traction Sandbox*'
+          mention: 'here'
+          mention_if: 'failure'
+          channel: '#ditp-gha-notifications'
+          url: ${{ secrets.ROCKETCHAT_WEBHOOK }}
+          commit: true
+          token: ${{ secrets.ROCKETCHAT_TOKEN }}
```
Comment (WadeBarnes) [inline .github/workflows/uninstall_install_sandbox.yaml:92]: `token: ${{ secrets.ROCKETCHAT_TOKEN }}` is not required if the `secrets.ROCKETCHAT_WEBHOOK` contains the token, which in our case it does.  Refer to your previous work with this action here; https://github.com/bcgov/von-bc-registries-audit/blob/main/.github/workflows/actions/deploy/action.yml#L83-L93

File: .github/workflows/uninstall_install_sandbox.yaml [hunk 3]
```diff
@@ -28,15 +28,28 @@ jobs:
...
           helm uninstall traction-sandbox -n ${{ secrets.OPENSHIFT_NAMESPACE }} --wait --timeout=20m
 
       - name: Remove Traction Openshift Objects
-        continue-on-error: true 
+        continue-on-error: false 
         run: |
           oc delete -n ${{ secrets.OPENSHIFT_NAMESPACE }} all,secret,pod,networkpolicy,configmap,pvc --selector "app.kubernetes.io/instance"=traction-sandbox
 
+      - name: Rocket.Chat Notification
+        uses: RocketChat/Rocket.Chat.GitHub.Action.Notification@master
```
Comment (WadeBarnes) [inline .github/workflows/uninstall_install_sandbox.yaml:41]:
```suggestion
        uses: RocketChat/Rocket.Chat.GitHub.Action.Notification@1.1.1
```
It's best practice to always target a released version of an action, that way you are not surprised by unexpected changes as you would be when targeting the `master` branch.  We covered this during your previous work with this action here; https://github.com/bcgov/von-bc-registries-audit/blob/main/.github/workflows/actions/deploy/action.yml#L83-L93

File: .github/workflows/uninstall_install_sandbox.yaml [hunk 4]
```diff
@@ -58,9 +71,22 @@ jobs:
...
         run: |
           helm repo add traction https://bcgov.github.io/traction 
           helm upgrade --install traction-sandbox traction/traction \
           -f ./helm-values/traction/values-sandbox.yaml \
-          -n ${{ secrets.OPENSHIFT_NAMESPACE }} --wait --timeout=7m
\ No newline at end of file
+          -n ${{ secrets.OPENSHIFT_NAMESPACE }} --wait --timeout=7m
+
+      - name: Rocket.Chat Notification
+        uses: RocketChat/Rocket.Chat.GitHub.Action.Notification@master
```
Comment (WadeBarnes) [inline .github/workflows/uninstall_install_sandbox.yaml:82]:
```suggestion
        uses: RocketChat/Rocket.Chat.GitHub.Action.Notification@1.1.1
```
It's best practice to always target a released version of an action, that way you are not surprised by unexpected changes as you would be when targeting the `master` branch.  We covered this during your previous work with this action here; https://github.com/bcgov/von-bc-registries-audit/blob/main/.github/workflows/actions/deploy/action.yml#L83-L93

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
Overall this workflow is triggering on a schedule (as expected) and on any commits to the `main` branch regardless of whether the changes have anything to do with the traction instance.  This will cause unexpected and unwanted refreshes to the traction instance.  A filter should be added to ensure the action is only triggered when changes are directly related to the traction instance.

Also see the specific comments below.

**Comment by WadeBarnes** [comment]
Example of path filters; https://github.com/hyperledger/indy-node/blob/main/.github/workflows/Push.yaml

**Comment by rajpalc7** [comment]
Thanks for the feedback wade, made the changes accordingly.

File: .github/workflows/uninstall_install_sandbox.yaml [hunk 5]
```diff
@@ -27,18 +30,29 @@ jobs:
...
+      #   run: |
+      #     helm uninstall traction-sandbox -n ${{ secrets.OPENSHIFT_NAMESPACE }} --wait --timeout=20m
 
-      - name: Remove Traction Openshift Objects
-        continue-on-error: true 
-        run: |
-          oc delete -n ${{ secrets.OPENSHIFT_NAMESPACE }} all,secret,pod,networkpolicy,configmap,pvc --selector "app.kubernetes.io/instance"=traction-sandbox
+      # - name: Remove Traction Openshift Objects
+      #   run: |
+      #     oc delete -n ${{ secrets.OPENSHIFT_NAMESPACE }} all,secret,pod,networkpolicy,configmap,pvc --selector "app.kubernetes.io/instance"=traction-sandbox
```
Comment (WadeBarnes) [inline .github/workflows/uninstall_install_sandbox.yaml:39]: Did you mean to comment out the `Uninstall Traction via Helm` and `Remove Traction Openshift Objects` steps?

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
Some comments below

**Comment by WadeBarnes** [comment]
That looks better @rajpalc7.  Please clean up the commit history.

**Comment by rajpalc7** [comment]
@WadeBarnes -  Commit history is cleaned up now

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
Hey @rajpalc7, I'm a little confused over the changes made by the cleanup.  The previous commits were just updates to the one `.github/workflows/uninstall_install_sandbox.yaml` file.  With the most recent commit there are changes to two other files included and it's showing `.github/workflows/uninstall_install_sandbox.yaml` is being added as a new file.

What happened?

**Comment by rajpalc7** [comment]
@WadeBarnes that might be because of the git HEAD reset command .  It should have been git reset HEAD~6 instead of HEAD~7 . Let me know if you want me to create another PR just for this file

**Comment by rajpalc7** [comment]
https://github.com/bcgov/trust-over-ip-configurations/pull/159/commits - I was able to create a new PR for review

**Comment by WadeBarnes** [comment]
The changes associated to this PR are being performed over here now; https://github.com/bcgov/trust-over-ip-configurations/pull/159

**Comment by WadeBarnes** [comment]
This PR is being closed in favor of https://github.com/bcgov/trust-over-ip-configurations/pull/159

**Pull request closed without merging**